cpp-client/deephaven/client/include/private/deephaven/client/server/server.h

```diff
@@ -67,6 +67,63 @@ struct ServerResponseHolder final : public CompletionQueueCallback {
   Response response_;
 };
 
+// Options for the creation of the underlying grpc channel.
+class ClientOptions {
+public:
+  typedef std::vector<std::pair<std::string, int>> int_options_t;
+  typedef std::vector<std::pair<std::string, std::string>> string_options_t;
+public:
+  static ClientOptions defaults() {
+    return ClientOptions();
+  }
+
+  ClientOptions &setUseTls(const bool useTls) {
+    useTls_ = useTls;
+    return *this;
+  }
+
+  ClientOptions &setPem(const std::string &pem) {  // root certificate to use
+    pem_ = pem;
+    return *this;
+  }
+
+  // See https://grpc.github.io/grpc/cpp/group__grpc__arg__keys.html for
+  // a list of available options.
+  // Example:
+  //   copts.setStringOption("grpc.min_reconnect_backoff_ms", 2000)
+  ClientOptions &setIntOption(const std::string &opt, const int val) {
+    intOptions_.emplace_back(opt, val);
+    return *this;
+  }
+
+  // See https://grpc.github.io/grpc/cpp/group__grpc__arg__keys.html for
+  // a list of available options.
+  // Example:
+  //   copts.setStringOption("grpc.target_name_override", "idonthaveadnsforthishost");
+  ClientOptions &setStringOption(const std::string &opt, const std::string &val) {
+    stringOptions_.emplace_back(opt, val);
+    return *this;
+  }
+
+  bool useTls() const { return useTls_; }
+  const std::string &pem() const { return pem_; }
+  const int_options_t &intOptions() const { return intOptions_; }
+  const string_options_t &stringOptions() const { return stringOptions_; }
+private:
+  ClientOptions()
+    : useTls_(false)
+    , pem_("")
+    , intOptions_()
+    , stringOptions_()
+  {}
+
+private:
+  bool useTls_;
```

Comment:
I'm slightly in favor of putting the defaults here, setting useTls_ specifically and leaving the others allone, as in:

```
  bool useTls_ = false;
  std::string pem_;
  int_options_t intOptions_;
  string_options_t stringOptions_;
```

And then making a constructor and destructor declaration here:

~~~
ClientOptions();
~ClientOptions();
~~~

and then in server.cc adding

```
ClientOptions::ClientOptions() = default;
ClientOptions::~ClientOptions() = default;
```

(the rationale for moving the definitions outline, into the cc file, is that the constructor code is "big enough" not to be inline)

Reply:
We don't need to define a destructor, right?  The one that we get implicitly defined is right, and unlike the constructor, we don't need (actually it won't work if) the destructor is private.

Reply:
If we don't define a destructor, the implicitly-defined one will be inlined. My personal style is to explicitly define a destructor (even if that definition is `ClientOptions::~ClientOptions() = default`) so that it is only expressed in one place in the code rather than sprinkled around many places as inlined code.

Reply:
Ok, for this kind of object that makes a lot of sense.